### PR TITLE
Fallback to checking eth node for missing reorg reverse blocks

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: pallets/cash/src/internal/events.rs [hunk 1]
```diff
@@ -248,12 +248,19 @@ pub fn memorize_chain_blocks<T: Config>(blocks: &ChainBlocks) -> Result<(), Reas
...
+    chain_id: ChainId,
+    hash: ChainHash,
+    starport: ChainStarport,
+) -> Result<ChainBlock, Reason> {
     let key = format!("cash::memorize_chain_blocks::{}", hash);
     let krf = StorageValueRef::persistent(key.as_bytes());
     match krf.get::<ChainBlock>() {
         Some(Some(block)) => Ok(block),
-        _ => Err(Reason::MissingBlock),
+        _ => match fetch_chain_block_by_hash(chain_id, hash, starport) {
```
Comment (hayesgm) [inline pallets/cash/src/internal/events.rs:295]: Here we go, the actual line. Looks great.

**Review by hayesgm — APPROVED** [review]
Great improvement

File: pallets/cash/src/lib.rs [hunk 2]
```diff
@@ -738,6 +738,11 @@ impl<T: Config> Module<T> {
     /// Initializes the set of supported assets from a config value.
     fn initialize_assets(assets: Vec<AssetInfo>) {
         for asset in assets {
+            log!("Adding assset {:?}", asset);
```
Comment (coburncoburn) [inline pallets/cash/src/lib.rs:768]: assset  with an an extra "s"

**Review by coburncoburn — APPROVED** [review]
great un-blocker.

**Pull request merged**